utils/global-state-update-gen/src/generic.rs

```diff
@@ -291,13 +291,21 @@ fn create_or_update_bid<T: StateReader>(
     pub_key: &PublicKey,
     recipient: &SeigniorageRecipient,
 ) {
+    let check_bid = |bid: &Bid| {
+        let bid_delegators: BTreeMap<_, _> = bid
+            .delegators()
+            .iter()
+            .map(|(key, delegator)| (key.clone(), *delegator.staked_amount()))
+            .collect();
+        &bid_delegators == recipient.delegator_stake() && *bid.staked_amount() == *recipient.stake()
+    };
     if state
         .get_bids()
         .get(pub_key)
-        .and_then(|bid| bid.total_staked_amount().ok())
-        == recipient.total_stake()
+        .map(check_bid)
+        .unwrap_or(false)
```

Comment:
Isn't this equivalent to simply `state.get_bids().get(pub_key) == bid_delegators` ?

Reply:
Not quite, because 1) types wouldn't check out, and 2) we're also checking the stake of the validator themselves, not only the delegators.